matchbox-server/src/main/java/ch/ahdis/matchbox/interceptors/MappingLanguageInterceptor.java

```diff
@@ -165,34 +165,10 @@ private FhirVersionEnum extractFhirVersion(final String header) {
 	public String compileAndSerializeMap(final String mapText,
 													 final FhirVersionEnum fhirVersion,
 													 final @Nullable String structureMapId) throws IOException {
-		final IWorkerContext workerContext;
-		if (this.matchboxEngineSupport.getClientContext().getOnlyOneEngine()) {
-			// In 'onlyOneEngine' mode, we cannot initialize a new engine for the compilation
-			if (FhirVersionEnum.R5.getFhirVersionString().equals(this.matchboxEngineSupport.getClientContext().getFhirVersion())) {
-				// We try to reuse the main engine if it's R5
-				workerContext = MatchboxEngineSupport.mainEngine.getContext();
-			} else {
-				// Otherwise, we have to initialize a new worker context
-				workerContext = MatchboxEngine.createR5WorkerContext();
-			}
-		} else {
-			// Let's get or initialize a new R5 engine
-			workerContext = MatchboxEngineSupport.mainEngine.getContextForFhirVersion(fhirVersion.getFhirVersionString());
-		}
-
-		final var scu = new MatchboxStructureMapUtilities(workerContext,
-																		  new TransformSupportServices(workerContext,
-																												 Collections.emptyList()),
-																		  MatchboxEngineSupport.mainEngine);
-		final StructureMap mapR5 = scu.parse(mapText, "map");
-		mapR5.getText().setStatus(Narrative.NarrativeStatus.GENERATED);
-		mapR5.getText().setDiv(new XhtmlNode(NodeType.Element, "div"));
-		final String render = StructureMapUtilities.render(mapR5);
-		mapR5.getText().getDiv().addTag("pre").addText(render);
+		StructureMap mapR5 = MatchboxEngineSupport.mainEngine.parseMapR5(mapText);
```

Comment:
`MatchboxEngine.parseMapR5()` doesn't handle the `onlyOneEngine` mode correctly, and should probably not. I think it's cleaner to handle that in matchbox-server.

Reply:
as dicussed: use createR5WorkerContext if we are not in a 5.0.0 context

Reply:
updated as discuessed.